### PR TITLE
Switch mechanics, indent

### PR DESCRIPTION
**Pull request by bounceme**
also now that I know about cindent i'm curious if there are other places like this to remove code and increase performance

### CODE REVIEW

**Comment by amadeus** [comment]
Testing this out, I am noticing some weirdness

If I just use the indentation given to me, I get this scenario:

```js
switch (test) {
	case 1:
	test = 2;
	word = 3;
	hi = 4;
	break;
	case 2:
	weird = 2;
	hi = 3;
	break;
	default:
	test = 10;
	word = 20;
	break;
}
```

Yet if I try to indent the first line, 2 different things  will happen:

```js
switch (test) {
	case 1:
		test = 2; // purposely indented this line
	word = 3;
	hi = 20;
	break;
	case 2:
		hi = 2; // Purposely indented this line
		weird = 2;
		strange = 3;
		break;
	default:
		weird = 3; // Purposely indented this line
		hi = 2;
		break;
}
```

**Comment by bounceme** [comment]
Your example is actually broken for me too. If I understand right, this is being caused by something to do with the breaks

**Comment by bounceme** [comment]
This seems to work now and has removed even more code

File: indent/javascript.vim [hunk 1]
```diff
@@ -475,11 +449,7 @@ function GetJavascriptIndent()
 
   " If the previous line ended with a block opening, add a level of indent.
   if s:Match(lnum, s:block_regex)
-    if (line =~ s:expr_case)
-      return indent(s:GetMSL(lnum, 0)) + s:sw()/2
-    else
       return indent(s:GetMSL(lnum, 0)) + s:sw()
```
Comment (davidchambers) [inline indent/javascript.vim]: :arrow_left: :arrow_left:

**Comment by davidchambers** [comment]
I :heart: all the red in the diff!

**Comment by bounceme** [comment]
also people could customize this in a few ways:
```
set cino+=:0
```

**Comment by amadeus** [comment]
Unfortunately this appears to have many more issues.

First off, here's the default indents if I don't change anything:

```js
switch (case) {
    case 1:
        test = 1;
    another = 2;
    hi = 3;
case 2:
    anotherTest = 1;
hmmm = 3;
break;
case 4:
    test = 1;
weird = 3;
}
```

It gets worse if I try to fix the indent:

```js
switch (case) {
    case 1:
        test = 1;
        test2 = 2; // this line had to be fixed
        test = 3;
    break; // this line had to be un-indented
case 2: // this line autoindents BACK when I type the :
}
```

**Comment by amadeus** [comment]
Oh wait, hold on, I had the wrong PR, one sec

**Comment by amadeus** [comment]
Would you like to hold off on merging this to find some other improvements you can make? Or would prefer to file those as separate PRs?

**Comment by bounceme** [comment]
i'd prefer fixing whatever bugs come of this as they are found, which with my testing of this, isn't so worrying to prevent a merge.

**Pull request merged**